gameengine.js

```diff
@@ -24,8 +24,9 @@ class GameEngine
      * animation.
      * 
      * @param {any} ctx A reference to the game context.
+     * @param {any} blink A reference to blink
      */
-    init(bottomProjectionContext, middleProjectionContext, gameCtx)
+    init(bottomProjectionContext, middleProjectionContext, gameCtx, blink)
```

Comment:
You don't need to and should pass in a parameter for blink. The game already has a reference to all of the entities in entities[];

You can loop through it and find blink using a function like:

for (int i = 0; i < this.entities.length; i++)
{
    if (this.entites[i] instanceof Blink)
{
// save him and pass to camera object constuctor here
}
}